numba/tests/test_unicode.py

```diff
@@ -337,6 +349,42 @@ def test_find(self, flags=no_pyobj_flags):
                                  cfunc(a, substr),
                                  "'%s'.find('%s')?" % (a, substr))
 
+    def test_count(self):
+        pyfunc = count_usecase
+        cfunc = njit(pyfunc)
+
+        for s in UNICODE_EXAMPLES:
+            extras = ['', ' ', 'xx', s[::-1], s[:-2], s[3:], s, s + s]
+            for sub in [x for x in extras]:
+                self.assertEqual(pyfunc(sub, s),
```

Comment:
I believe `sub` and `s` are in the wrong order here. This also applies to the other tests.

Reply:
@esc Thanks for the code. I have tried this code and it passes all the tests except one where sub_string is empty. That part needs some change so I'll do that and resubmit it again.

Reply:
Oh, that is good, which test was failing? I didn't notice any failing tests. Did this happen after changing the order or `sub` and `s`.

Reply:
Yes it happened after changing the order for sub and s. It was failing for a case when the slice_range is bigger than src_len. But apart from that the algorithm is really useful because the same will be used in rfind method as well.